### PR TITLE
 fix(opentelemetry-instrumentation-aws-sdk): error when ReturnConsumedCapacity is set to None

### PR DESCRIPTION
**Pull request by pauloprestes**
## Which problem is this PR solving?

- Fixes https://github.com/open-telemetry/opentelemetry-js-contrib/issues/898

## Short description of the changes

- check if function map exists on ConsumedCapacity property.

## Checklist

- [ ] Ran `npm run test-all-versions` for the edited package(s) on the latest commit if applicable.

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/dynamodb.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class DynamodbServiceExtension implements ServiceExtension {
     const operation = response.request.commandName;
 
     if (operation === 'BatchGetItem') {
-      if ('ConsumedCapacity' in response.data) {
+      if ('ConsumedCapacity' in response.data && 'map' in response.data.ConsumedCapacity){
```
Comment (blumamir) [inline plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/dynamodb.ts:64]: Thanks for reporting it and creating a fix.
I think using the `in` operator to check if a function is found on the object is a bit non-ideomatic.
What do you think about changing it to: `if(Array.isArray(response.data?.ConsumedCapacity))`?
Also - is it possible to add a unit-test for this case?

Comment (pauloprestes) [reply plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/dynamodb.ts]: Thanks for your comment @blumamir . I've pushed some changes to address your comment. [667eade](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/899/commits/667eade433b7e9a49a756861b048b6ecbab9deef)

**Review by blumamir — APPROVED** [review]
LGTM, thanks again for the fix

**Comment by blumamir** [comment]
@pauloprestes please run `npm run lint:fix`, thanks

**Pull request merged**